lib/integrations/awsoidc/deployservice_iam_config.go

```diff
@@ -0,0 +1,328 @@
+		return trace.Wrap(err)
+	}
+
+	if req.AccountID == "" {
+		callerIdentity, err := clt.GetCallerIdentity(ctx, nil)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		req.AccountID = aws.ToString(callerIdentity.Account)
+	}
+
+	if err := createBoundaryPolicyForTaskRole(ctx, clt, req); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := createTaskRole(ctx, clt, req); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := addPolicyToTaskRole(ctx, clt, req); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := addPolicyToIntegrationRole(ctx, clt, req); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// createBoundaryPolicyForTaskRole creates a Policy to be used as TaskRole's Role Boundary.
+// It allows for the TaskRole to:
+// - connect to any RDS DB
+// - Get, Put and Delete Role Policies to manage the Policy Statements when adding other rds-db:connect entries
+// - write application logs to CloudWatch
+func createBoundaryPolicyForTaskRole(ctx context.Context, clt DeployServiceIAMConfigureClient, req DeployServiceIAMConfigureRequest) error {
+	taskRoleARN := awslib.RoleARN(req.partitionID, req.AccountID, req.TaskRole)
+
+	taskRoleBoundaryPolicyDocument, err := awslib.NewPolicyDocument(
+		awslib.StatementForRDSDBConnect(),
+		awslib.StatementForIAMEditRolePolicy(taskRoleARN),
+		awslib.StatementForWritingLogs(),
+	).Marshal()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = clt.CreatePolicy(ctx, &iam.CreatePolicyInput{
+		PolicyName:     &req.TaskRoleBoundaryPolicyName,
+		PolicyDocument: &taskRoleBoundaryPolicyDocument,
+		Tags:           req.ResourceCreationTags.ToIAMTags(),
+	})
+	if err != nil {
+		if trace.IsAlreadyExists(awslib.ConvertIAMv2Error(err)) {
+			return trace.AlreadyExists("Policy %q already exists, please remove it and try again.", req.TaskRoleBoundaryPolicyName)
+		}
+		return trace.Wrap(err)
+	}
+
+	log.Printf("TaskRole: Boundary Policy %q created.\n", req.TaskRoleBoundaryPolicyName)
+	return nil
+}
+
+// createTaskRole creates the TaskRole and sets up the Role Boundary and its Trust Relationship.
+func createTaskRole(ctx context.Context, clt DeployServiceIAMConfigureClient, req DeployServiceIAMConfigureRequest) error {
+	taskRoleAssumeRoleDocument, err := awslib.NewPolicyDocument(
+		awslib.StatementForECSTasksAssumeRole(),
+	).Marshal()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	policyARNForRoleBoundary := awslib.PolicyARN(req.partitionID, req.AccountID, req.TaskRoleBoundaryPolicyName)
+
+	_, err = clt.CreateRole(ctx, &iam.CreateRoleInput{
+		RoleName:                 &req.TaskRole,
+		AssumeRolePolicyDocument: &taskRoleAssumeRoleDocument,
+		PermissionsBoundary:      &policyARNForRoleBoundary,
+		Tags:                     req.ResourceCreationTags.ToIAMTags(),
+	})
+	if err != nil {
+		if trace.IsAlreadyExists(awslib.ConvertIAMv2Error(err)) {
+			return trace.AlreadyExists("Role %q already exists, please remove it and try again.", req.TaskRole)
+		}
+		return trace.Wrap(err)
```

Comment:
I wanted to change the message so that it includes an action ("remove it").
If I return the converted type, then it will be the raw AWS error message.

Reply:
i meant the `return trace.Wrap(err)` after the already exist check (like `return trace.Wrap(the-converted-error)`. I guess not big difference.

Reply:
Oh right :pray: 
Yes! changing